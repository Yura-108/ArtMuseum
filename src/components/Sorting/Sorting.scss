@import '@styles/_mixins.scss';
@import '@styles/_variables.scss';

.sorting {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 10px;
  span {
    @include font-style('Lexend Deca', 400, 18px, 24px);
    color: $color-text-grey;
  }
  select {
    background: transparent;
    outline: none;
    @include font-style('Lexend Deca', 400, 16px, 24px);
    color: $color-text-grey;
    padding: 5px;
    margin-left: 5px;
  }
  @media (max-width: $width-main-container) {
    justify-content: center;
  }
}
