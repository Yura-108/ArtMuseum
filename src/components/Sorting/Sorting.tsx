import React from 'react';
import './Sorting.scss';
import { SortMethod } from '@types/SortMethod.ts';
import { SortingProps } from '@types/componentsPropsTypes.ts';

const Sorting: React.FC<SortingProps> = ({ setSortMethod }) => {
  const handleSortChange = (method: SortMethod) => {
    setSortMethod(method);
  };

  return (
    <div className={'sorting'}>
      <span>Sorting:</span>
      <select
        name="sortingSelector"
        id="sortingSelector"
        onChange={(e) => handleSortChange(e.target.value as SortMethod)}
      >
        <option value="title">By title</option>
        <option value="artist">By artist</option>
        <option value="date">By date</option>
      </select>
    </div>
  );
};

export default Sorting;
