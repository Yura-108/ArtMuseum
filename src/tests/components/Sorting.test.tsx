import { render, screen, fireEvent } from '@testing-library/react';
import Sorting from '@components/Sorting/Sorting.tsx';


describe('Sorting', () => {
  const mockSetSortMethod = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the Sorting component', () => {
    render(<Sorting setSortMethod={mockSetSortMethod} />);

    expect(screen.getByText('Sorting:')).toBeInTheDocument();
    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByRole('option', { name: 'By title' })).toBeInTheDocument();
    expect(screen.getByRole('option', { name: 'By artist' })).toBeInTheDocument();
    expect(screen.getByRole('option', { name: 'By date' })).toBeInTheDocument();
  });

  it('calls setSortMethod when a sort method is selected', () => {
    render(<Sorting setSortMethod={mockSetSortMethod} />);

    const selectElement = screen.getByRole('combobox');
    fireEvent.change(selectElement, { target: { value: 'artist' } });

    expect(mockSetSortMethod).toHaveBeenCalledTimes(1);
    expect(mockSetSortMethod).toHaveBeenCalledWith('artist');
  });

  it('updates value when a new option is selected', () => {
    render(<Sorting setSortMethod={mockSetSortMethod} />);

    const selectElement = screen.getByRole('combobox');
    fireEvent.change(selectElement, { target: { value: 'date' } });

    expect(selectElement).toHaveValue('date');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Sorting setSortMethod={mockSetSortMethod} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
