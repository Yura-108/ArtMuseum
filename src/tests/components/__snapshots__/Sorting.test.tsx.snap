// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sorting"
  >
    <span>
      Sorting:
    </span>
    <select
      id="sortingSelector"
      name="sortingSelector"
    >
      <option
        value="title"
      >
        By title
      </option>
      <option
        value="artist"
      >
        By artist
      </option>
      <option
        value="date"
      >
        By date
      </option>
    </select>
  </div>
</DocumentFragment>
`;
